### PR TITLE
Workaround for the TiDB date bug

### PR DESCRIPTION
**Pull request by tirsen**

### CODE REVIEW

File: misk-hibernate/src/main/kotlin/misk/vitess/DockerTidbCluster.kt [hunk 1]
```diff
@@ -31,11 +29,12 @@ class TidbCluster(
   fun openConnection(): Connection = dataSource().connection
 
   private fun dataSource(): DriverDataSource {
-    val jdbcUrl = config.withDefaults().buildJdbcUrl(
-        Environment.TESTING)
+    val config = config.withDefaults().copy(database = "")
+    val jdbcUrl = config.buildJdbcUrl(Environment.TESTING)
+    println("jdbcUrl = ${jdbcUrl}")
```
Comment (swankjesse) [inline misk-hibernate/src/main/kotlin/misk/vitess/DockerTidbCluster.kt]: nit: debug

File: misk-hibernate/src/main/kotlin/misk/vitess/DockerTidbCluster.kt [hunk 2]
```diff
@@ -31,11 +29,12 @@ class TidbCluster(
...
     return DriverDataSource(
-        jdbcUrl, config.type.driverClassName, Properties(),
-        config.username, config.password)
+        jdbcUrl, this.config.type.driverClassName, Properties(),
+        "root", "")
```
Comment (swankjesse) [inline misk-hibernate/src/main/kotlin/misk/vitess/DockerTidbCluster.kt]: really?

File: misk-hibernate/src/main/kotlin/misk/vitess/DockerTidbCluster.kt [hunk 3]
```diff
@@ -159,7 +159,7 @@ class DockerTidbCluster(
 
     waitUntilHealthy()
     cluster.openConnection().use { c ->
-      c.createStatement().executeUpdate("SET GLOBAL time_zone = '+00:00'")
+      c.prepareStatement("SET GLOBAL time_zone = '+00:00'").executeUpdate()
```
Comment (swankjesse) [inline misk-hibernate/src/main/kotlin/misk/vitess/DockerTidbCluster.kt]: dig

File: misk-hibernate/src/test/kotlin/misk/hibernate/TransacterTest.kt [hunk 4]
```diff
@@ -34,10 +34,6 @@ abstract class TransacterTest {
   fun happyPath() {
     createTestData()
 
-    if (hasDateBug()) {
```
Comment (swankjesse) [inline misk-hibernate/src/test/kotlin/misk/hibernate/TransacterTest.kt]: nice

**Comment by chris-ryan-square** [comment]
If this PR is still relevant, then please fix conflicts, rebase and get a green CI build.
If the PR is no longer relevant please close.

This PR will be closed at the end of January 2021 otherwise.

**Comment by chris-ryan-square** [comment]
Closing request as per previous announcement.

**Pull request closed without merging**